src/MicrosoftTeams.ts

```diff
@@ -2329,3 +2329,90 @@ export function getChatMembers(
   const messageId = sendMessageRequest(parentWindow, "getChatMembers");
   callbacks[messageId] = callback;
 }
+
+/**
+ * Namespace to interact with the conversational subEntities inside the tab
+ */
+export namespace conversations {
+
+  /**
+   * @private
+   * Hide from docs.
+   * ------
+   */
+  export interface ConversationalSubEntity {
+
+    /**
+     * A function that is called once the conversation Id has been created.
+     */
+    onStartConversation?: (conversationId: string) => void;
+
+    /**
+     * A function that is called if the pane is closed
+     */
+    onCloseConversation?: () => void;
+  }
+
+  /**
+   * @private
+   * Hide from docs
+   * --------------
+   * Allows the user to start a conversation with each subentity inside a tab
+   * @param subEntityId The Id of the subEntity where the conversation is taking place
+   * @param title The title of the conversation
+   * @param conversationalSubEntity Callback containing the conversation Id and if the tab pane was closed
+   */
+  export function startConversation(
+    subEntityId: string,
+    title: string,
+    conversationalSubEntity: ConversationalSubEntity
+  ): void {
+    const messageId = sendMessageRequest(parentWindow, "startConversation", [
+      subEntityId,
+      title
+    ]);
+    callbacks[messageId] = (conversationId: string) => {
+      if (conversationId) {
+        conversationalSubEntity.onStartConversation(conversationId);
+      } else {
+        conversationalSubEntity.onCloseConversation();
+      }
+    };
+  }
+
+  /**
+   * @private
+   * Hide from docs
+   * --------------
+   * Allows the user to show the conversation in the right pane
+   * @param subEntityId The Id of the subEntity where the conversation is taking place
+   * @param title The title of the conversation
+   * @param conversationId The Id of the conversation
+   * @param conversationalSubEntity Callback containing if the tab pane was closed
+   */
+  export function showConversation(
+    subEntityId: string,
+    title: string,
+    conversationId: string,
+    conversationalSubEntity: ConversationalSubEntity
+  ): void {
+    const messageId = sendMessageRequest(parentWindow, "showConversation", [
```

Comment:
What happens when two or more showConversations are called executed without a close?  Just thinking developers might put code in the onCloseConversation and are depending on that being called if a new conversation it opened.  Would we also need to provide the conversationId and subEntityId on the onClose callback also?

Reply:
Just thinking of the example where a user clicks the "X" button and we fire the onCloseConversation.  Maybe in my app I want to do some visual thing like change an icon state or even do some logging.  So maybe having that information in the onClose would help?

Reply:
- We expect two or more conversations to be just called using showConversations. When we receive consecutive calls, we would treat it as the previous conversation has been closed. And the pane would be opened with the new conversation. This is so that the developer doesnt need to call the close if a user just opens a new conversation.

- The onclose would be just to tell us that the right pane doesnt need to be kept open. Hence subEntityId or conversation Id is not required. Thats why I removed it.

- The developer would know as soon as the  onCloseConversation callback is fired. This is fired only for the two cases - X and opening the tab pane. That should be enough if the developer wants to take some action. I can pass in a string or something. Will run this by Loki